Prog/Voices_q15_recode.sas

```diff
@@ -55,73 +55,326 @@ data Q15_recode_unq;
+			nightlife=nightlife+index(Q15_text,"Going To The Clubs");
+			nightlife=nightlife+index(Q15_text,"Good Night Life");
+
+
+			services=index(Q15_text,"Free Help Services");
+			services=services+index(Q15_text,"So Many Resources");
+			services=services+index(Q15_text,"Access To Good Medical");
+			services=services+index(Q15_text,"Access To Most Household Services");
+			services=services+index(Q15_text,"Access To My Karate");
+			services=services+index(Q15_text,"Access To Resources");
+			services=services+index(Q15_text,"Access To Top Medical");
+			services=services+index(Q15_text,"Full Social Security"); *was in safety; 
+
+			entertain=index(Q15_text,"Access To Recreational Facilities");
+			entertain=entertain+index(Q15_text,"Access To Recreational Spots");
+			entertain=entertain+index(Q15_text,"Access To Recreation Zone");
+			if Q15_text="A Lot Of Amusment Parks" then entertain=1;
+			if Q15_text="Bars, Clubs, Music Venues, Variety in Dining" then entertain=1; *was community;
+			if Q15_text in("A Lot Of Things To Do" "A Lot To Do" "All Of The Family Activities Available" "Always Lots To Do" "Entertainment Experience" "It Has Lots Of Stuff To Do"
+							"Boating" "Many Things To Do" "Lots Of Free Things To Do" "Lots Of Fun Stuff For Kids" "Lots Of Stuff To Do" "Lots Of Things To Do" "Lots To Do" "Lots To See"
+							"Places To Go" "Places To See" "Plenty Of Things To Do" "So Many Things To Do" "So Much To Do" "The Attractions" "There Is Entertainment Spritz Thouhout The Dmv"
+							"Things To Do" "Things To See" "Tons Of Things To Do" "Lot To Do" "Movies") then entertain=1; *was not matched;
+
+			metro=index(Q15_text,"Metro Accesibility");
+			metro=metro+index(Q15_text,"Metro Accessiblity");
+			if Q15_text in("Easy Metro Access" "Metro Access" ) then metro=1;
+			if Q15_recode='' and Q15_Text in("Metro" "Metro Accessible") then metro=1;
+
+			transpo=index(Q15_text,"Public Transit"); *was metro;
+			if Q15_text in("Nice Transit" "Mass Transit" "Transit" "Easy Travel & Transit" "Calles" "Decent Transporation Links" "Easy To Get Around" "Easy To Travel To Other Places"
+							"Transportstio" "Travel Options" "It's Easy To Get Around") then transpo=1;
+
+			people=index(Q15_text,"Affluent Communities");
+			people=people+index(Q15_text,"Citizens Are Highly Educated"); 
+			people=people+index(Q15_text,"People With Money"); 
+
+			if culture > 0 then Q15_recode="CULTURE";
+			if politics > 0 then Q15_recode="POLITICS";
+			if capital > 0 then Q15_recode="CAPITAL";
+			if monuments > 0 then Q15_recode="MONUMENTS";
+			if jobs > 0 then Q15_recode="JOBS";
+			if diversity > 0 then Q15_recode="DIVERSITY";
+			if food > 0 then Q15_recode="FOOD";
+			if nightlife > 0 then Q15_recode="NIGHTLIFE";
+			if services > 0 then Q15_recode="SERVICES";
+			if entertain > 0 then Q15_recode="ENTERTAINMENT";
+			if people > 0 then Q15_recode="PEOPLE";
+			if transpo > 0 then Q15_recode="TRANSPORTATION";
+			if metro > 0 then Q15_recode="METRO";
```

Comment:
It's possible that responses could have multiple matches, in which case the matching category that comes latest in this list would be the one kept. Is that what you want?

If there is a hierarchy, I would suggest using `if...then...; else if ... then...;` listing responses in descending order of priority. That way, once it find a match, it stops checking.

Reply:
@ptatian yes - added. seems to have fixed most of the above things which were coded but not picking up.